exercises/proverb/canonical-data.json

```diff
@@ -0,0 +1,98 @@
+                    "expected": [
+                        "And all for the want of a nail."
+                    ],
+                    "input": [
+                        "nail"
+                    ],
+                    "property": "recite"
+                },
+                {
+                    "description": "two pieces",
+                    "expected": [
+                        "For want of a nail the shoe was lost.",
+                        "And all for the want of a nail."
+                    ],
+                    "input": [
+                        "nail",
+                        "shoe"
+                    ],
+                    "property": "recite"
+                },
+                {
+                    "description": "three pieces",
+                    "expected": [
+                        "For want of a nail the shoe was lost.",
+                        "For want of a shoe the horse was lost.",
+                        "And all for the want of a nail."
+                    ],
+                    "input": [
+                        "nail",
+                        "shoe",
+                        "horse"
+                    ],
+                    "property": "recite"
+                },
+                {
+                    "description": "full proverb",
+                    "expected": [
+                        "For want of a nail the shoe was lost.",
+                        "For want of a shoe the horse was lost.",
+                        "For want of a horse the rider was lost.",
+                        "For want of a rider the message was lost.",
+                        "For want of a message the battle was lost.",
+                        "For want of a battle the kingdom was lost.",
+                        "And all for the want of a nail."
+                    ],
+                    "input": [
+                        "nail",
+                        "shoe",
+                        "horse",
+                        "rider",
+                        "message",
+                        "battle",
+                        "kingdom"
+                    ],
+                    "property": "recite"
+                },
+                {
+                    "description": "four pieces modernized",
+                    "expected": [
+                        "For want of a pin the gun was lost.",
+                        "For want of a gun the soldier was lost.",
+                        "For want of a soldier the battle was lost.",
+                        "And all for the want of a pin."
+                    ],
+                    "input": [
+                        "pin",
+                        "gun",
+                        "soldier",
+                        "battle"
+                    ],
+                    "property": "recite"
+                }
+            ],
+            "description": "Construct a proverb of the form 'for want of an X, the Y was lost'"
```

Comment:
I wonder if someone will miss this at the crucial time because it's on the bottom. But I don't think it's necessary at all. I would remove it and remove one level of `cases` nesting.

Reply:
The description does say that the lines of the proverb could vary, only the "input".